### PR TITLE
ROC-3255 Retry requests

### PR DESCRIPTION
**Pull request by timja-kainos**

### CODE REVIEW

File: src/main/app/claims/claimStoreClient.ts [hunk 1]
```diff
@@ -89,20 +91,21 @@ export class ClaimStoreClient {
...
         }
-        if (claim) {
-          return new Claim().deserialize(claim)
-        } else {
-          throw new Error('Call was successful, but received an empty claim instance')
-        }
+        return new Claim().deserialize(claim.body)
+      })
+      .catch(err => {
+        console.log(err)
```
Comment (robertparkinson) [inline src/main/app/claims/claimStoreClient.ts]: is this a WIP? if not this needs removing

Comment (timja-kainos) [reply src/main/app/claims/claimStoreClient.ts]: lint will block that dw ;), its wip but pretty much done, just been discussing with damian and rafal

File: package.json [hunk 2]
```diff
@@ -77,6 +77,7 @@
     "reflect-metadata": "^0.1.10",
     "request": "^2.81.0",
     "request-promise-native": "^1.0.3",
+    "requestretry": "^1.13.0",
```
Comment (geek-soft) [inline package.json]: We need to use forked version here.

File: src/main/app/claims/claimStoreClient.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { request } from 'client/request'
+import { retryingRequest } from 'client/request'
```
Comment (geek-soft) [inline src/main/app/claims/claimStoreClient.ts]: Didn't we agree to use retrying request by default in all places?

File: config/default.json [hunk 4]
```diff
@@ -71,6 +71,9 @@
   "http": {
     "timeout": 10000
   },
+  "requestRetry": {
+    "maxAttempts": 5
```
Comment (geek-soft) [inline config/default.json]: I would say 3 would be enough. Retrying 5 times on most HTTP errors (status code >= 300) seems pointless.

File: src/main/app/client/request.ts [hunk 5]
```diff
@@ -3,27 +3,34 @@ import { RequestTracingHandler } from 'logging/requestTracingHandler'
...
-
+const defaultOptions = {
+  json: true,
+  timeout: timeout,
+  fullResponse: false,
+  maxAttempts: maxAttempts
+}
 const request: RequestPromiseAPI = RequestTracingHandler.proxy(
-  RequestLoggingHandler.proxy(rpWithDefaultsSet)
+  RequestLoggingHandler.proxy(requestPromise.defaults(defaultOptions))
```
Comment (geek-soft) [inline src/main/app/client/request.ts]: Passing `maxAttempts: maxAttempts` to regular request object does not make much sense and might be misleading.

File: src/main/app/claims/claimStoreClient.ts (hunk 3 above)
Comment (rlewan) [reply src/main/app/claims/claimStoreClient.ts]: As agreed it will be first applied to claim store calls only and then to the rest.

Comment (geek-soft) [reply src/main/app/claims/claimStoreClient.ts]: I am happy with using `retryingRequest` for now to emphasise different behaviour in claim-store client. Later when we apply retrying request to all other places we should rename it back to be `request`

**Pull request merged**